packages/react/src/hooks/contracts/usePrepareContractWrite.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  FetchSignerResult,
+  PrepareWriteContractConfig,
+  PrepareWriteContractResult,
+  prepareWriteContract,
+} from '@wagmi/core'
+import { hashQueryKey } from 'react-query'
+
+import { QueryConfig, QueryFunctionArgs } from '../../types'
+import { useSigner } from '../accounts'
+import { useChainId, useQuery } from '../utils'
+
+export type UsePrepareContractWriteArgs = Omit<
+  PrepareWriteContractConfig,
+  'signerOrProvider'
+>
+export type UsePrepareContractWriteConfig = QueryConfig<
+  PrepareWriteContractResult,
+  Error
+>
+
+export const queryKey = (
+  {
+    args,
+    addressOrName,
+    contractInterface,
+    functionName,
+    overrides,
+  }: UsePrepareContractWriteArgs,
+  { chainId, signer }: { chainId?: number; signer?: FetchSignerResult },
```

Comment:
Is the first object the public values and second object the internal values?

Reply:
Yeah!